iface/interface.go

```diff
@@ -139,88 +127,78 @@ func (iface *ifacePlugin) GetMetricTypes(cfg plugin.ConfigType) ([]plugin.Metric
 
 // CollectMetrics returns list of requested metric values
 // It returns error in case retrieval was not successful
-func (iface *ifacePlugin) CollectMetrics(metricTypes []plugin.MetricType) ([]plugin.MetricType, error) {
-	err := iface.setProcPath(metricTypes[0])
+func (iface *IfaceCollector) CollectMetrics(mts []plugin.Metric) ([]plugin.Metric, error) {
+	err := iface.setProcPath(mts[0].Config)
 	if err != nil {
 		return nil, err
 	}
 
-	metrics := []plugin.MetricType{}
+	metrics := []plugin.Metric{}
 
-	if err := iface.getStats(iface.stats); err != nil {
+	if err := iface.getStats(); err != nil {
 		return nil, err
 	}
 	curTime := time.Now()
-	for _, metricType := range metricTypes {
-		ns := metricType.Namespace()
+	for _, mt := range mts {
+		ns := mt.Namespace
 		if len(ns) < 5 {
 			return nil, fmt.Errorf("Namespace length is too short (len = %d)", len(ns))
 		}
 		if ns[len(ns)-2].Value == "*" {
 			for itf, istats := range iface.stats {
 				val := getMapValueByNamespace(istats.(map[string]interface{}), ns.Strings()[4:])
 				if val != nil {
-					ns1 := core.NewNamespace(createNamespace(itf, ns[len(ns)-1].Value)...)
-					interfaceName := ns[len(ns)-2].Name
-					ns1[len(ns1)-2].Name = interfaceName
-					interfaceConfiguration, err := getInterfaceConfiguration(interfaceName)
+					ns1 := plugin.NewNamespace(createNamespace(itf, ns[len(ns)-1].Value)...)
```

Comment:
please use copyNamespace here. Currently we are losing information about dynamic element of namespace